### ISSUE TITLE
TOM admin interface timing out when loading a DataProduct

### ISSUE THREAD

**Issue by rachel3834** [TOMToolkit/tom_base#930]
**Describe the bug**
When using MOP's admin interface to review uploaded DataProducts, I clicked on the link to an example data product, but the page failed to load and eventually timed out.  

Looking at the stern logs I see this:

mop-ffbf8dbd8-jkmpt nginx 10.100.26.43 - - [15/May/2024:16:41:25 +0000] "GET /admin/tom_dataproducts/dataproduct/83486/change/ HTTP/1.1" 499 0 "https://mop.lco.global/admin/tom_dataproducts/dataproduct/" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0" "207.71.206.53"
mop-ffbf8dbd8-jkmpt mop [2024-05-15 16:41:33 +0000] [1] [ERROR] Worker (pid:8) was sent SIGKILL! Perhaps out of memory?

Its not clear to me why this should be an issue, unless perhaps we're having issues with MOP's resources?

**Comment by jchate6** [comment]
Just to clarify, is the issue with the link to the data from the admin interface here? 

![Image](https://github.com/TOMToolkit/tom_base/assets/32683393/9a3016f5-3161-474d-a461-ee378e8c83dd)

**Comment by rachel3834** [comment]
Yes, it seems to hang when asked to display fields that have many-to-many
relationships; possibly this is because we have >10,000 targets so this
could be quite a computationally intensive view to generate?
